### PR TITLE
Refactor mk_dataset to add XArray

### PR DESCRIPTION
**Pull request by sashakames**

### CODE REVIEW

File: pkg/esgcet/mk_dataset.py [hunk 1]
```diff
@@ -259,139 +256,58 @@ def get_file(self, mapdata, fn_trid):
...
+    def set_variables(self, record, scanobj):
+        variables = self.get_variables(scanobj)
+        # use the correct facet id string to get the variable if pre-specified in the record
+        vid = record[self.variable_name]
+        if vid in variables:
+            var_rec = variables[vid]
+            if "long_name" in var_rec:
+                record["variable_long_name"] = var_rec["long_name"]
+            elif "info" in var_rec:
+                record["variable_long_name"] = var_rec["info"]
```
Comment (nocollier) [inline pkg/esgcet/mk_dataset.py:268]:
```suggestion
                record["variable_info"] = var_rec["info"]
```

Comment (nocollier) [reply pkg/esgcet/mk_dataset.py:268]: Or was this intentional?

Comment (nocollier) [reply pkg/esgcet/mk_dataset.py:268]: Also, this section could be written in a more easily extensible loop.
```
        if vid in variables:
            var_rec = variables[vid]
            record[self.variable_name] = vid
            if self.variable_name == "variable_id":
                record["variable"] = vid
            for attr in ["long_name","info","standard_name","units"]:
                record[f"variable_{attr}"] = var_rec[attr] 
```

**Review by nocollier — COMMENTED** [review]
I do not yet understand everything going on in here, but it looks good apart from the one question I raised. I will dig in and look at how this all works now.

File: pkg/esgcet/mk_dataset.py (hunk 1 above)
Comment (sashakames) [reply pkg/esgcet/mk_dataset.py:268]: `variable_long_name` is the standard ESGF search field we need to populate, also `cf_standard_name`.  So we are stuck with particularly formatted keys, also some files use `long_name` and others use `info` instead.  This is one area where ESGF adopted a standard where there may not have been anything so strict.

Comment (sashakames) [reply pkg/esgcet/mk_dataset.py:268]: Of course, there's no documentation (too often I've been in a hurry) so that would improve these things...

**Comment by sashakames** [comment]
@nocollier Just pushed an update, feel free to take a look again as you see fit

**Pull request merged**